controllers/resource/reconciler.go

```diff
@@ -117,6 +117,8 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 		}
 		resources = append(resources, r...)
 	case anywherev1.CloudStackDatacenterKind:
+		// max len = len(workers) + CP + etcd
+		spec.CloudStackMachineConfigs = make(map[string]*anywherev1.CloudStackMachineConfig, len(cs.Spec.WorkerNodeGroupConfigurations)+2)
```

Comment:
The template builder makes use of the `CloudStackMachineConfigs`, so making sure to add them to the template builder here.